Userland/Libraries/LibGfx/AntiAliasingPainter.cpp

```diff
@@ -169,3 +170,218 @@ void Gfx::AntiAliasingPainter::draw_cubic_bezier_curve(const FloatPoint& control
         draw_line(fp1, fp2, color, thickness, style);
     });
 }
+
+void Gfx::AntiAliasingPainter::draw_circle(IntPoint center, int radius, Color color)
+{
+    /*
+    Algorithm from: https://cs.uwaterloo.ca/research/tr/1984/CS-84-38.pdf
+    Inline comments are from the paper.
+    */
+
+    // TODO: Generalize to ellipses (see paper)
+
+    constexpr int subpixel_resoulution = 255;
+    const int intensity { color.alpha() };
+
+    // Note: Variable names below are from the paper
```

Comment:
meh.
if you know what they are, just give them a better name.

Reply:
I've renamed most of them & I think it's a clearer now, I've kept the delta/derivative ones mostly the same as I can't think of a much better name.